### PR TITLE
Update DiscoveredCluster spec and remove clusters_mgmt api

### PR DESCRIPTION
**Pull request by JakobGray**
User story: https://github.com/open-cluster-management/backlog/issues/10770

The DiscoveredCluster spec included extra information not being used
so all unused fields have been removed. With this change, all data
gathered from OCM can be obtained through the subscription endpoint.
All things related to the clusters_mgmt api have been removed. The data
used to mock subscriptions has been updated to include full objects

### CODE REVIEW

**Review by nathanweatherly — APPROVED** [review]

File: api/v1/discoveredcluster_types.go [hunk 1]
```diff
@@ -27,27 +27,14 @@ import (
...
+	Status            string       `json:"status,omitempty" yaml:"status,omitempty"`
 	IsManagedCluster  bool         `json:"isManagedCluster,omitempty" yaml:"isManagedCluster,omitempty"`
 
 	ProviderConnections []corev1.ObjectReference `json:"providerConnections"`
-
-	Subscription SubscriptionSpec `json:"subscription" yaml:"subscription"`
-}
-
-type SubscriptionSpec struct {
-	Status       string `json:"status" yaml:"status"`
```
Comment (zkayyali812) [inline api/v1/discoveredcluster_types.go:47]: Actually this value is read in the UI here - 
https://github.com/open-cluster-management/console/blob/main/frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx#L47

Should we coordinate PRs?

File: api/v1/discoveredcluster_types.go [hunk 2]
```diff
@@ -27,27 +27,14 @@ import (
...
 	CreationTimestamp *metav1.Time `json:"creation_timestamp,omitempty" yaml:"creation_timestamp,omitempty"`
 	ActivityTimestamp *metav1.Time `json:"activity_timestamp,omitempty" yaml:"activity_timestamp,omitempty"`
-	OpenshiftVersion  string       `json:"openshiftVersion" yaml:"openshiftVersion"`
-	Region            string       `json:"region,omitempty" yaml:"region,omitempty"`
+	OpenshiftVersion  string       `json:"openshiftVersion,omitempty" yaml:"openshiftVersion,omitempty"`
 	CloudProvider     string       `json:"cloudProvider,omitempty" yaml:"cloudProvider,omitempty"`
-	HealthState       string       `json:"healthState,omitempty" yaml:"healthState,omitempty"`
-	State             string       `json:"state,omitempty" yaml:"state,omitempty"`
-	Product           string       `json:"product,omitempty" yaml:"product,omitempty"`
+	Status            string       `json:"status,omitempty" yaml:"status,omitempty"`
```
Comment (zkayyali812) [inline api/v1/discoveredcluster_types.go:34]: Is this supposed to be `subscription.status`?

File: api/v1/discoveredcluster_types.go (hunk 1 above)
Comment (zkayyali812) [reply api/v1/discoveredcluster_types.go:47]: This should also be updated - https://github.com/open-cluster-management/console/blob/main/frontend/src/resources/discovered-cluster.ts#L18

File: api/v1/discoveredcluster_types.go (hunk 2 above)
Comment (JakobGray) [reply api/v1/discoveredcluster_types.go:34]: Yes

**Pull request merged**